### DISCUSSION TITLE
MovieWriter Play/Pause/Stop function

### DISCUSSION THREAD

**Discussion by MR2011** [godotengine/godot-proposals#6965] [Engine Core]
Currently the MovieWriter starts recording as soon as the application starts and stops when the application is closed (at least that's how I understand it).

I would like to have the option to manually start, stop and pause the recording in my application code. This allows the user to set up the application state as desired without recording unnecessary things that need to be removed in a video editor afterwards.

I read the note that the MovieWriter is not designed for the use by end users to record videos while playing. However, I don't need realtime recording and the MovieWriter allows to render videos at a higher resolution than the screen resolution, so I prefer using the MovieWriter itself.

What do you think?

**Comment by AThousandShips** [comment]
What's preventing you from setting up the specific things from script or just the start state in the editor? The movie writer mode isn't just a minor change it depends on flags set on startup, so don't think it's very easy or convenient to set up that state once the game is running

**Reply by MR2011** [reply]
The idea in my app is that the user configures the desired state through the UI and then runs a simulation. I want to export this simulation as a video. Of course I can save this state and then open a second app instance with the desired configuration to start the video recording but this is only a workaround.

**Reply by AThousandShips** [reply]
Okay! That does sound outside of the intended scope, sorry wasn't clear to me that you were intending an app that has this functionality

**Comment by Calinou** [comment]
Unfortunately, the way Movie Maker mode is enabled makes it difficult to toggle at run-time. It expects to be using its own AudioServer (to write audio data without any distortion), and it also forces non-realtime simulation. https://github.com/godotengine/godot/pull/63458 and https://github.com/godotengine/godot/pull/60284 may help with this, but they're not sufficient in themselves to make Movie Maker mode toggleable at run-time.

As a workaround, use `OS.execute()` with `OS.get_executable_path()` to start a second instance of the process with Movie Maker mode enabled, with the original instance remaining as-is. If you want to quit the original instance while recording, use `OS.set_restart_on_exit()` to specify MovieWriter command line arguments, then call `get_tree().quit()`. While in the second instance that writes the movie, use `OS.set_restart_on_exit()` again to restart the original process without any command line arguments.

**Reply by MR2011** [reply]
Thank you @Calinou I had the same idea with starting a second instance. For now I'm using the png saving method you showed in https://github.com/Calinou/godot-video-rendering-demo
